server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlExecParser.g4

```diff
@@ -1744,32 +1748,34 @@ dbs_triggered_sql_statement_adv: dbs_call | dbs_delete | dbs_get_diagnostics_sta
 dbs_triggered_sql_statement_basic: dbs_triggered_sql_statement;
 dbs_version_id: dbs_host_variable | dbs_sql_identifier | dbs_string_constant;
 dbs_view_name: dbs_host_variable | dsb_alias_name;
-dbs_pieceSize : T=dbs_sql_identifier {validateTokenWithRegex($T.text, "\\d+[MmGgKk]", "db2SqlParser.pieceSize");};
+dbs_pieceSize : dbs_sql_identifier;
 dbs_comma_separator: (COMMASEPARATORDB2 | COMMACHAR);
 dbs_semicolon_end: SEMICOLON_FS | SEMICOLONSEPARATORSQL;
 
 ////////  BUILDING BLOCKS   ///
 ///////  Ref: https://www.ibm.com/docs/en/db2-for-zos/12?topic=elements-naming-conventions ////
 dbs_integer_constant: INTEGERLITERAL | NUMERICLITERAL; //range 1 - 32767
-dbs_sql_identifier: IDENTIFIER | DSNDB04 | TRANSACTION | RECORDS | dbs_special_name;
+dbs_sql_identifier: dbs_generic_identifier_without_inbuild_function_names | dbs_inbuild_functions;
+dbs_generic_identifier_without_inbuild_function_names : IDENTIFIER | JOIN | UNION| EXCEPT | INTERSECT | dbs_special_name ;
 dbs_constant : (dbs_string_constant | dbs_integer_constant);
 dbs_generic_name: dbs_host_variable | dbs_string_constant | dbs_sql_identifier; //TODO : check this
 dbs_string_constant: CHAR_STRING_LITERAL | HEXSTRING | BXSTRING | GRAPHIC_CONSTANT | DATELITERAL;
 dbs_accelerator_name: dbs_sql_identifier; // TODO {1-8 chars}
-dsb_alias_name: dbs_sql_identifier DOT_FS dbs_sql_identifier DOT_FS dbs_sql_identifier
-			| dbs_sql_identifier DOT_FS dbs_sql_identifier
-			| dbs_sql_identifier;
+dsb_alias_name
+    : dbs_sql_identifier (DOT_FS dbs_sql_identifier (DOT_FS dbs_sql_identifier)?)?
+    ;
 dbs_array_type_name: dbs_sql_identifier (DOT_FS dbs_sql_identifier)?;
 dbs_authorization_name: dbs_sql_identifier;
 dbs_authorization_id: dbs_string_constant;
 dbs_aux_table_name: dsb_alias_name; //TODO {validateLength($T.text, "Auxiliary table name", 128);};
 dbs_table_name: dsb_alias_name;
 //dbs_bpname: ('B' | 'b') ('P' |' p') INTEGERLITERAL (('K' | 'k') INTEGERLITERAL)?;
 dbs_collection_id: dbs_sql_identifier;
-dbs_column_name: (dsb_alias_name DOT_FS)?  dbs_sql_identifier; //TODO {validateLength($T.text, "Column name", 30);};
+//dbs_column_name: (dsb_alias_name DOT_FS)?  IDENTIFIER;
+dbs_column_name: dbs_sql_identifier (DOT_FS  (dbs_sql_identifier (DOT_FS dbs_sql_identifier)?)?  dbs_sql_identifier)?; //TODO {validateLength($T.text, "Column name", 30);};
```

Comment:
dbs_column_name in this form cannot be used in the context of e.g. ALTER TABLE statement.